### PR TITLE
Dev: Add a verify check for some functions to always have a compile time value

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
Ready!

File: src/libasr/intrinsic_func_registry_util_gen.py [hunk 1]
```diff
@@ -321,6 +321,10 @@ def gen_verify_args(func_name):
     global src
     src += indent + R"static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x, diag::Diagnostics& diagnostics) {" + "\n"
     add_verify_arg_type_src(func_name)
+    if func_name in skip_fn_instantiation:
+        src += indent * 2 + 'ASRUtils::require_impl(x.m_value, '\
+            f'"Missing compile time value, `{func_name}` intrinsic output can '\
```
Comment (certik) [inline src/libasr/intrinsic_func_registry_util_gen.py:326]:
```suggestion
            f'"Missing compile time value, `{func_name}` intrinsic output must '\
```
?

**Review by certik — APPROVED** [review]
Should we also rename `skip_fn_instantiation` to `compile_time_only_fn`?

This is good to merge as is, just some ideas.

File: src/libasr/intrinsic_func_registry_util_gen.py (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply src/libasr/intrinsic_func_registry_util_gen.py:326]: Yes

**Pull request merged**